support/collect/collection.go

```diff
@@ -0,0 +1,1334 @@
+package collect
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"math/rand"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type Collection[T any] struct {
+	items []T
+}
+
+func (c *Collection[T]) After(value T) *T {
+	for i, item := range c.items {
+		if reflect.DeepEqual(item, value) && i+1 < len(c.items) {
+			return &c.items[i+1]
+		}
+	}
+	return nil
+}
+
+func (c *Collection[T]) All() []T {
+	return c.items
+}
+
+func (c *Collection[T]) Average(keyFunc func(T) float64) float64 {
+	if len(c.items) == 0 {
+		return 0
+	}
+	return c.Sum(keyFunc) / float64(len(c.items))
+}
+
+func (c *Collection[T]) Avg(keyFunc func(T) float64) float64 {
+	return c.Average(keyFunc)
+}
+
+func (c *Collection[T]) Before(value T) *T {
+	for i, item := range c.items {
+		if reflect.DeepEqual(item, value) && i > 0 {
+			return &c.items[i-1]
+		}
+	}
+	return nil
+}
+
+func (c *Collection[T]) Chunk(size int) [][]T {
+	if size <= 0 {
+		return [][]T{}
+	}
+
+	var chunks [][]T
+	for i := 0; i < len(c.items); i += size {
+		end := i + size
+		if end > len(c.items) {
+			end = len(c.items)
+		}
+		chunks = append(chunks, c.items[i:end])
+	}
+	return chunks
+}
+
+func (c *Collection[T]) ChunkWhile(predicate func(T, int, []T) bool) [][]T {
+	if len(c.items) == 0 {
+		return [][]T{}
+	}
+
+	var chunks [][]T
+	var currentChunk []T
+
+	for i, item := range c.items {
+		if len(currentChunk) == 0 || predicate(item, i, currentChunk) {
+			currentChunk = append(currentChunk, item)
+		} else {
+			chunks = append(chunks, currentChunk)
+			currentChunk = []T{item}
+		}
+	}
+
+	if len(currentChunk) > 0 {
+		chunks = append(chunks, currentChunk)
+	}
+
+	return chunks
+}
+
+func (c *Collection[T]) Clone() *Collection[T] {
+	cloned := make([]T, len(c.items))
+	copy(cloned, c.items)
+	return &Collection[T]{items: cloned}
+}
+
+func (c *Collection[T]) Collapse() *Collection[T] {
+	var flattened []T
+	for _, item := range c.items {
+		v := reflect.ValueOf(item)
+		if v.Kind() == reflect.Slice {
+			for i := 0; i < v.Len(); i++ {
+				if elem, ok := v.Index(i).Interface().(T); ok {
+					flattened = append(flattened, elem)
+				}
+			}
+		} else {
+			flattened = append(flattened, item)
+		}
+	}
+	return &Collection[T]{items: flattened}
+}
+
+func Collect[T any](items []T) *Collection[T] {
```

Comment:
How about changing `collect.Collect` to `collect.Of`? This way, it avoids redundancy and becomes more concise.